floodlight/core/events.py

```diff
@@ -61,3 +95,148 @@ def custom(self):
         )
 
         return custom
+
+    @property
+    def essential_missing(self):
+        missing_columns = [
+            col for col in essential_events_columns if col not in self.essential
+        ]
+
+        if not missing_columns:
+            return None
+        else:
+            return missing_columns
+
+    @property
+    def essential_invalid(self):
+        incorrect_columns = [
+            col
+            for col in self.essential
+            if not self.column_values_in_range(col, essential_events_columns)
+        ]
+
+        if not incorrect_columns:
+            return None
+        else:
+            return incorrect_columns
+
+    @property
+    def protected_missing(self):
+        missing_columns = [
+            col for col in protected_columns if col not in self.protected
+        ]
+
+        if not missing_columns:
+            missing_columns = None
+
+        return missing_columns
+
+    @property
+    def protected_invalid(self):
+        incorrect_columns = [
+            col
+            for col in self.protected
+            if not self.column_values_in_range(col, protected_columns)
+        ]
+        if not incorrect_columns:
+            incorrect_columns = None
+
+        return incorrect_columns
+
+    def column_values_in_range(self, col: str, definitions: Dict[str, Dict]) -> bool:
+        """Perform the checks for value range for a single column of the inner event
+        DataFrame using the specifications from floodlight.core.definitions.
+
+        Parameters
+        ----------
+        col: str
+            Column of the inner event DataFrame to be checked
+        definitions: Dict
+            Dictionary for each column name to be checked. Each entry is a Dictionary
```

Comment:
Same as below. I think it's really hard to explain this syntax in prose. Maybe just pseudocode the syntax, or give an example, or refer to the definitions.py.